state/contract.go

```diff
@@ -29,16 +29,18 @@ type (
 
 	contract struct {
 		*Account
-		dirtyCode  bool      // contract's code has been set
-		dirtyState bool      // contract's account state has changed
-		code       []byte    // contract byte-code
-		trie       trie.Trie // storage trie of the contract
+		dirtyCode  bool        // contract's code has been set
+		dirtyState bool        // contract's account state has changed
+		addr       hash.PKHash // contract's address
+		code       []byte      // contract byte-code
+		trie       trie.Trie   // storage trie of the contract
 	}
 )
 
 // GetState get the value from contract storage
 func (c *contract) GetState(key hash.Hash32B) ([]byte, error) {
-	v, err := c.trie.Get(key[:])
+	slot := hash.Hash256b(append(key[:], c.addr[:]...))
```

Comment:
maybe we can reverse the order to append(c.addr[:], key...). It has not difference in functionality, and may be clearer as addr.key